frappe/desk/desktop.py

```diff
@@ -305,6 +303,8 @@ def _in_active_domains(item):
 						new_item['is_query_report'] = 1
 					else:
 						new_item['ref_doctype'] = report.get('ref_doctype')
+				elif item.type == "Route":
+					new_item['route'] = frappe.db.get_value('Route', item.link_to, 'route')
```

Comment:
The problem with having arbitrary routes is that role permission manager won't apply. In the future, system managers will be able to modify desk globally, in that case, showing routes that can't be accessed won't be a good experience.

I see the example has link to a **New** form for a DocType. If that is the case,  we can have a route option in the child table itself. Where we can hold a checkbox. So if the Link To field is set to DocType, and if it's not a child doctype or a single doctype, we can show a checkbox `Route to New Form`